Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp

```diff
@@ -145,16 +145,30 @@ static std::unique_ptr<WPE::DRM::Connector> chooseConnector(int fd, drmModeRes*
 
 static std::unique_ptr<WPE::DRM::Crtc> chooseCrtcForConnector(int fd, drmModeRes* resources, const WPE::DRM::Connector& connector, GError** error)
 {
+    /* Try the currently connected encoder+crtc */
     WPE::DRM::UniquePtr<drmModeEncoder> encoder(drmModeGetEncoder(fd, connector.encoderID()));
-    if (!encoder) {
-        g_set_error_literal(error, WPE_DISPLAY_ERROR, WPE_DISPLAY_ERROR_CONNECTION_FAILED, "Failed to get the connector encoder");
-        return nullptr;
+    if (encoder) {
+        for (int i = 0; i < resources->count_crtcs; ++i) {
+            if (resources->crtcs[i] == encoder->crtc_id) {
+                WPE::DRM::UniquePtr<drmModeCrtc> drmCrtc(drmModeGetCrtc(fd, resources->crtcs[i]));
+                return WPE::DRM::Crtc::create(fd, drmCrtc.get(), i);
+            }
+        }
     }
 
-    for (int i = 0; i < resources->count_crtcs; ++i) {
-        if (resources->crtcs[i] == encoder->crtc_id) {
-            WPE::DRM::UniquePtr<drmModeCrtc> drmCrtc(drmModeGetCrtc(fd, resources->crtcs[i]));
-            return WPE::DRM::Crtc::create(fd, drmCrtc.get(), i);
+    /* If no active crtc was found, pick the first possible crtc, then try the first possible for crtc */
+    for (int i = 0; i < resources->count_encoders; ++i) {
+        WPE::DRM::UniquePtr<drmModeEncoder> encoder(drmModeGetEncoder(fd, resources->encoders[i]));
+        if (!encoder)
+            continue;
+
+        for (int j = 0; j < resources->count_crtcs; ++j) {
+            const uint32_t crtcMask = 1 << j;
+            const uint32_t crtcId = resources->crtcs[j];
```

Comment:
Move this inside the if, or even better remove it and just use resources->crtcs[j].